src/org.hdfgroup.hdfview/hdf/view/TableView/DataProviderFactory.java

```diff
@@ -470,7 +481,46 @@ private void updateAtomicValue(Object bufObject, Object newValue, int bufIndex)
                 Array.setDouble(bufObject, bufIndex, dvalue);
                 break;
             default:
-                Array.set(bufObject, bufIndex, newValue);
+                String rname = bufObject.getClass().getSimpleName();
+                log.trace("updateAtomicValue(): getSimpleName={}", rname);
+                switch (rname.charAt(0)) {
+                case 'B':
+                    Byte bValue = Byte.valueOf((String) newValue);
```

Comment:
Note that these cases don't check for a NumberFormatException from Byte.valueOf, Short.valueof, etc. and might cause bad behavior in the TableView if someone types an invalid value for the datatype

Reply:
In tests. the DataValidator checked for correct number formats, before getting here.